### DISCUSSION TITLE
Text Stroke with Text Anti-Alias?

### DISCUSSION THREAD

**Discussion by a-darya** [ImageMagick/ImageMagick#3417] [Help]
Hi,

Stroke does not appear to work when disabling text anti alias. Is there a reason for this? Using the ImageMagick C++ library ( ImageMagick 7.0.10-50).

Thanks.

**Comment by fmw42** [comment]
I can confirm in command line using IM 7.0.11.3 Q16 Mac OSX.

magick -size 500x200 -background gray -fill red -font arial -gravity center label:"This is a test" label0.png

![label0](https://user-images.githubusercontent.com/13458218/111705325-bb131e00-87fd-11eb-9e7c-51317b42b2e3.png)


magick -size 500x200 -background gray -fill red -stroke red -strokewidth 3 -font arial -gravity center label:"This is a test" label1.png

![label1](https://user-images.githubusercontent.com/13458218/111705344-c1a19580-87fd-11eb-9a70-999453f46419.png)


magick -size 500x200 -background gray -fill red -stroke red -strokewidth 3 -font arial -gravity center +antialias label:"This is a test" label2.png

![label2](https://user-images.githubusercontent.com/13458218/111708224-af762600-8802-11eb-9eec-459918056877.png)

**Reply by a-darya** [reply]
To disable/turn off antialias via command line, the argument seems to be '+antialias' rather than '-antialias' (https://legacy.imagemagick.org/Usage/antialiasing/).

Here is the command tested and the result:
aarch64-linux-gnu-magick -size 500x200 -background gray -fill red -stroke black -strokewidth 3 -font Default.ttf -gravity center +antialias label:"This is a test" strokeTest.png

![strokeTest](https://user-images.githubusercontent.com/80797607/111707803-f9124100-8801-11eb-830c-e6e82ad0a74f.png)

With anti alias enabled, no issues:
aarch64-linux-gnu-magick -size 500x200 -background gray -fill red -stroke black -strokewidth 3 -font Default.ttf -gravity center -antialias label:"This is a test" strokeTest2.png

![strokeTest2](https://user-images.githubusercontent.com/80797607/111708043-5c03d800-8802-11eb-98bf-0e4967f82c14.png)

Output of the version is:
Version: ImageMagick 7.0.10-50 Q8 aarch64 2020-12-18 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): freetype jng jpeg png zlib

**Reply by a-darya** [reply]
I've updated to the latest version and the issue still persists

Version: ImageMagick 7.0.11-3 Q8 aarch64 2021-03-06 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5)
Delegates (built-in): freetype jng jpeg png zlib

**Comment by snibgo** [comment]
> Stroke does not appear to work when disabling text anti alias.

To disable anti-aliasing at the CLI, use "+antialias".

This works fine for me with IM v7.0.8-64. As fmw42 says, please show your C++ code.

**Reply by fmw42** [reply]
My typo. I have corrected my post to confirm the issue.

**Reply by a-darya** [reply]
Since I'm having an issue with the CLI as well (posted above), I think this may be a problem with the version I'm using
